src/graphics/PlanningView.ts

```diff
@@ -483,88 +335,168 @@ export function drawHealthBarAboveHead(unitIndex: number, underworld: Underworld
-          if (manaAfterSpell < u.manaMax) {
-            const hurtX = manaBarHurtProps.x + config.UNIT_UI_BAR_WIDTH / zoom * manaAfterSpell / u.manaMax;
-            globalThis.unitOverlayGraphics.drawRect(
-              // Show the manaBarHurtBar on the right side of the mana  bar
-              hurtX,
-              manaBarHurtProps.y,
-              // Special width calculation required to prevent overfillmana
-              // from rendering wrongly 
-              manaBarProps.width - (hurtX - manaBarProps.x),
-              manaBarHurtProps.height);
+          const manaAfterPrediction = predictionUnit.mana;
+          if (manaAfterPrediction < u.mana) {
+            globalThis.unitOverlayGraphics.beginFill(colors.manaDarkBlue, 1.0);
+          }
+          else {
+            globalThis.unitOverlayGraphics.beginFill(colors.manaBrightBlue, 1.0);
           }
-          // if (manaAfterSpell > u.mana) {
-          //   globalThis.unitOverlayGraphics?.beginFill(manaBarHealColor, 1.0);
-          // }
+
+          let fillRect = getFillRect(u, 0, manaBarMax, u.mana, manaAfterPrediction, zoom);
+          globalThis.unitOverlayGraphics.drawRect(
+            fillRect.x,
+            fillRect.y,
+            fillRect.width,
+            fillRect.height);
         }
+        globalThis.unitOverlayGraphics.endFill();
       }
-      globalThis.unitOverlayGraphics?.endFill();
     }
 
   }
 }
+
+export function getFillRect(unit: Unit.IUnit, min: number, max: number, value1: number, value2: number, zoom: number): { x: number, y: number, width: number, height: number } {
+  const start01 = Math.max(0, Math.min(value1 / (min + max), 1))
+  const end01 = Math.max(0, Math.min(value2 / (min + max), 1))
+
+  const widthAdjusted = config.UNIT_UI_BAR_WIDTH / zoom;
+  const fillWidth = widthAdjusted * Math.abs(end01 - start01);
+  const height = config.UNIT_UI_BAR_HEIGHT / zoom;
+
+  return {
+    x: unit.x - widthAdjusted / 2 + Math.min(start01, end01) * widthAdjusted,
+    // - height so that bar stays in the same position relative to the unit
+    // regardless of zoom
+    // - config.HEALTH_BAR_UI_Y_POS so that it renders above their head instead of
+    // on their center point
+    y: unit.y - config.HEALTH_BAR_UI_Y_POS * (unit.image?.sprite.scale.y || 1) - height,
+    width: fillWidth,
+    height,
+  }
+}
+
+export function drawUnitMarker(imagePath: string, pos: Vec2, extraScale: number = 1) {
+  const zoom = getCamera().zoom;
+  // 1/zoom keeps the attention marker the same size regardless of the level of zoom
+  // Math.sin makes the attention marker swell and shink so it grabs the player's attention
+  // + 1 makes it go from 0 to 2 instead of -1 to 1
+  // / 8 limits the change in size
+  const markerScale = ((1 / zoom) + (Math.sin(Date.now() / 500) + 1) / 8) * extraScale;
+  const markerHeightHalf = 16 * markerScale;
+  const markerMarginAboveHealthBar = 10;
+
+  // TODO - move marker higher if the unit is a miniboss
```

Comment:
You should be able to use unit.image.scale.y to determine this since minibosses and stacked summoned units are just scaled up.

Reply:
Another TODO I forgot to remove haha. Had to jump around a lot and do intermittent testing with these changes. Will clean it up shortly.